eng/scripts/Remove-WormStorageAccounts.ps1

```diff
@@ -0,0 +1,38 @@
+# Helper script for removing storage accounts with WORM that sometimes get leaked from live tests not set up to clean
+# up their resource policies
+
+[CmdletBinding(SupportsShouldProcess=$True)]
+param(
+    [string]$GroupPrefix
+)
+
+# Be a little defensive so we don't delete non-live test groups via naming convention
+if (!$groupPrefix -or !$GroupPrefix.StartsWith('rg-')) {
+    Write-Error "The -GroupPrefix parameter must start with 'rg-'"
+    exit 1
+}
+
+$groups = Get-AzResourceGroup | ? { $_.ResourceGroupName.StartsWith($GroupPrefix) } | ? { $_.ProvisioningState -ne 'Deleting' }
+
+foreach ($group in $groups) {
+    $accounts = Get-AzStorageAccount -ResourceGroupName $group.ResourceGroupName
+    if ($accounts) {
+        foreach ($account in $accounts) {
+            if ($PSCmdlet.ShouldProcess($group.ResourceGroupName)) {
+                Write-Host "Removing $($account.StorageAccountName) in $($account.ResourceGroupName)"
+            } else {
+                Write-Host "What if: Removing $($account.StorageAccountName) in $($account.ResourceGroupName)"
+            }
+            $ctx = New-AzStorageContext -StorageAccountName $account.StorageAccountName
+            $ctx | Get-AzStorageContainer | Get-AzStorageBlob | Remove-AzStorageBlob -Force
+            # Use AzRm cmdlet as deletion will only work through ARM with the immutability policies defined on the blobs
+            $ctx | Get-AzStorageContainer | % { Remove-AzRmStorageContainer -Name $_.Name -StorageAccountName $ctx.StorageAccountName -ResourceGroupName $group.ResourceGroupName -Force }
+            Remove-AzStorageAccount -StorageAccountName $account.StorageAccountName -ResourceGroupName $account.ResourceGroupName -Force
+        }
+    }
+    if ($PSCmdlet.ShouldProcess($group.ResourceGroupName)) {
+        Remove-AzResourceGroup -ResourceGroupName $group.ResourceGroupName -Force -AsJob
```

Comment:
I'm wrapping this instead of just using the native WhatIf because even with WhatIf it still farms it out to a job which can be misleading and make the user think the group is actually being deleted.